### PR TITLE
realize rowset remove api

### PR DESCRIPTION
**Pull request by kangpinghuang**

### CODE REVIEW

File: be/src/olap/rowset/alpha_rowset.cpp [hunk 1]
```diff
@@ -117,9 +136,13 @@ int64_t AlphaRowset::start_version() const {
     return _rowset_meta->version().first;
 }
 
-bool AlphaRowset::create_hard_links(std::vector<std::string>* success_links) {
+bool AlphaRowset::create_files_with_new_name(std::vector<std::string>* success_links) {
```
Comment (chaoyli) [inline be/src/olap/rowset/alpha_rowset.cpp]: make_snapshot will be better?

File: be/src/olap/rowset/alpha_rowset.cpp [hunk 2]
```diff
@@ -75,6 +90,10 @@ RowsetMetaSharedPtr AlphaRowset::rowset_meta() const {
 }
 
 void AlphaRowset::set_version(Version version) {
+    if (_removed) {
```
Comment (chaoyli) [inline be/src/olap/rowset/alpha_rowset.cpp]: I think Rowset is already removed in memory structure synchronously if _removed is true.
So this judgement will be redundant.

File: be/src/olap/rowset/alpha_rowset.cpp [hunk 3]
```diff
@@ -60,9 +68,16 @@ OLAPStatus AlphaRowset::copy(RowsetWriter* dest_rowset_writer) {
 }
 
 OLAPStatus AlphaRowset::remove() {
-    // TODO(hkp) : add delete code
-    // delete rowset from meta
-    // delete segment groups 
+    OlapMeta* meta = _data_dir->get_meta();
+    OLAPStatus status = RowsetMetaManager::remove(meta, rowset_id());
```
Comment (chaoyli) [inline be/src/olap/rowset/alpha_rowset.cpp]: If process core dump after remove rowset_meta, how to remove_files in segment_group?

**Review by chaoyli — CHANGES_REQUESTED** [review]

File: be/src/olap/rowset/alpha_rowset.cpp (hunk 3 above)
Comment (kangpinghuang) [reply be/src/olap/rowset/alpha_rowset.cpp]: file gc

**Pull request merged**